pkg/kube/vault/vault.go

```diff
@@ -430,17 +432,22 @@ func GetVaults(client kubernetes.Interface, vaultOperatorClient versioned.Interf
 	for _, v := range vaultList.Items {
 		vaultName := v.Name
 		vaultAuthSaName := GetAuthSaName(v)
-		vaultURL, err := services.FindServiceURL(client, ns, vaultName)
-		if err != nil {
-			log.Logger().Warnf("error finding vault service url setting to empty string, err: %s", err)
-			vaultURL = ""
-		}
-		if vaultURL == "" {
-			vaultURL, err = services.FindIngressURL(client, ns, vaultName)
+		// default to using internal kubernetes service dns name
+		vaultURL := fmt.Sprintf(defaultInternalVaultURL, vaultName)
+		// is there a better way to do this?
+		if os.Getenv("KUBERNETES_SERVICE_HOST") == "" {
```

Comment:
wonder if this is ewnough to check that we're running incluster seeing that we dont have the common options factory in these functions?

Reply:
It should be easy to refactor the `IsInCluster()` method off the factory and into a util method as it depends on precisely nothing